site/management.md

```diff
@@ -335,6 +348,61 @@ curl -i --header "authorization: Bearer &lt;token&gt;" http://localhost:15672/ap
+<pre class="lang-ini">
+management.oauth_scopes = "openid profile"
+</pre>
+This configuration implies that your authorization server must able to
+grant additional scopes otherwise having only these 2 scopes will not grant the user access to the management ui.
+
+### Minimum scope required and how the UI determines the username from the token
+
+RabbitMQ requires, at minimum, the `openid` scope. This is because RabbitMQ uses some claims in the *id token* to
+determine the username and display it on the top right corner of the management UI. The *id token* is sent by the
+Authorization server if we include the `openid` scope in the authorization request.
+
+RabbitMQ reads the `user_name` claim from the *id_token*. If it does not have it, RabbitMQ uses the `sub` claim
+instead.
+
+### Configure OpenID Connect Discovery endpoint
+
+By default, RabbitMQ assumes the OpenID Connect Discovery endpoint is at `<management.oauth_provider_url>/.well-known/openid-configuration`. If your endpoint differs, you can set yours via `management.oauth_metadata_url` setting.
+
+RabbitMQ uses this endpoint to discover other endpoints such as **token** endpoint, **logout** endpoint, and others.
+
+### Logout workflow
+
+RabbitMQ follows the [OpenID Connect RP-Initiated Logout 1.0 ](https://openid.net/specs/openid-connect-rpinitiated-1_0.html)
+specification to implement the logout workflow. This means that the logout workflow is triggered from the Management UI when we click on the **Logout** button. Logging out from RabbitMQ management UI effectively logs the user out from
+the management UI itself but also from the Identity Provider.
+
+There are other two additional scenarios which can trigger a logout. One scenario occurs when the OAuth Token expires. Although RabbitMQ renews the token on the background before it expires, should the token expired, the user is logged out.
+The second scenario is when the management UI session exceeds the maximum allowed time configured on the [Login Session Timeout](https://www.rabbitmq.com/management.html#login-session-timeout).
+
+### Special attention to CSP header `connect-src`
+
+In other to support OAuth 2.0 protocol, RabbitMQ has to make asynchronous REST calls to the [OpenId Connect Discovery endpoint](#Configure-OpenID-Connect-Discovery-endpoint). If we override the default [CSP headers](#csp), we have to make sure that the `connect-src` CSP directive whitelists the [OpenId Connect Discovery endpoint](#Configure-OpenID-Connect-Discovery-endpoint).
+
+For instance, if we configured the CSP header with the value `default-src 'self'` we are, by default, setting `connect-src 'self'` which means we are denying RabbitMQ access to any external endpoint; hence disabling OAuth 2.0.
+
+In addition to the `connect-src` CSP header, RabbitMQ also needs these two CSP directives otherwise the OAuth 2.0 functionality may not work: `unsafe-eval` `unsafe-inline`.
```

Comment:
In addition to the `connect-src` CSP header, RabbitMQ also needs the two CSP directives `unsafe-eval` and `unsafe-inline`, otherwise the OAuth 2.0 functionality may not work.